### DISCUSSION TITLE
How enable https ip ssl access

### DISCUSSION THREAD

**Discussion by huang-x-h** [apache/apisix#7702] [Q&A] [answered]
Now, we delopy apisix and access `http://ip:9080`, then enable ssl and access `https://ip:9443`

![1660651233668](https://user-images.githubusercontent.com/381860/184874559-152f44b1-8d2b-4d5c-b616-7bdf557b0109.png)

So how to enable https, when just access by ip not by domain.

Now, we use a nginx server to set ssl and proxy apisix to fix this 😭

**Comment by Lewisyixin** [comment, marked as answer]
how ip works with ssl.... shouldn't ssl just used for domain?

**Reply by huang-x-h** [reply]
Customer requirement... In the inner network, want to use https

**Reply by tokers** [reply]
As you can see the error is not related to SSL but **Connection Refused**. Check out proxy settings and firewall settings, or maybe the networking settings.

**Reply by huang-x-h** [reply]
> As you can see the error is not related to SSL but **Connection Refused**. Check out proxy settings and firewall settings, or maybe the networking settings.

No firewall setting, just access apisix server ssl port, maybe apisix will check the SSL Certificate is not invalid? because the certificate is create by self

**Reply by tzssangglass** [reply]
It looks like this is a network layer error. Is port 9443 open and APISIX listen on 9443?

**Reply by huang-x-h** [reply]
> It looks like this is a network layer error. Is port 9443 open and APISIX listen on 9443?

yes, use the 9443 port, the config.yaml file 

![1660803222639](https://user-images.githubusercontent.com/381860/185307357-1d91708c-9299-4a60-9e37-5cecb12ce4f8.png)

**Reply by tzssangglass** [reply]
I think you need to troubleshoot network issues first, such as starting an nginx and listening on port 9443 for SSL to see if you can access it. Because judging from the browser response, this is a network issue.

**Reply by huang-x-h** [reply]
Now, the problem is solved.

Add `fallback_sni` option

![image](https://user-images.githubusercontent.com/381860/185821865-343fbe81-81f3-46b9-b770-98409754be8d.png)

**Reply by tzssangglass** [reply]
`fallback_sni` is used for cases where SNI is not passed by client or LB

**Reply by tokers** [reply]
Show us your APISIX configuration, especially for the listening settings. Just talking is useless.

**Reply by huang-x-h** [reply]
> Show us your APISIX configuration, especially for the listening settings. Just talking is useless.

First, we upload ip SSL Certificate

![1661217167820](https://user-images.githubusercontent.com/381860/186046712-8ab45548-9597-4347-acad-f235f000e7a9.png)

Second, change `config.yaml` add custom ssl options

```
apisix:
  node_listen: 9080
ssl:
  listen_port: 9443
  fallback_sni: "10.7.1.16"
```

Now, we access `https://10.7.1.15:9443/`

![1661217584443](https://user-images.githubusercontent.com/381860/186047293-b732a068-1b72-489c-b8f1-c3789ae2d632.png)

if `fallback_sni` not set, we access the site will show

![1661217656845](https://user-images.githubusercontent.com/381860/186047423-b6b83591-1b3b-45e1-ab9b-f2463562d687.png)

**Reply by tokers** [reply]
That's a normal case; if I remember correctly, the browser won't take the SNI to APISIX if you don't use the domain to access it. Even if the browser does, the SNI and the IP address don't match. How did APISIX know which certificate it should return to you?

**Reply by huang-x-h** [reply]
Yes, so we add fallback_sni

"How did APISIX know which certificate it should return to you?" maybe it's here, https://github.com/apache/apisix/blob/master/apisix/ssl.lua#L39-L55

**Reply by tokers** [reply]
You said "if fallback_sni not set, we access the site will show xxx". What's your problem? It's expected.
